src/plugins/mod.rs

```diff
@@ -125,16 +131,16 @@ impl Manager {
         manager.register_generator::<transform::secrets::Plugin>(proj)?;
         manager.register_vault_generator(proj)?;
 
-        manager.register_transform::<transform::abs_path::Plugin>(proj)?;
-        manager.register_transform::<transform::default_tags::Plugin>(proj)?;
-        manager.register_transform::<transform::sources::Plugin>(proj)?;
-        manager.register_transform::<transform::secrets::Plugin>(proj)?;
-        manager.register_transform::<transform::remove_build::Plugin>(proj)?;
-        manager.register_vault_transform(proj)?;
+        manager.register_transform::<transform::abs_path::Plugin>(proj, subcommand)?;
+        manager.register_transform::<transform::default_tags::Plugin>(proj, subcommand)?;
+        manager.register_transform::<transform::sources::Plugin>(proj, subcommand)?;
+        manager.register_transform::<transform::secrets::Plugin>(proj, subcommand)?;
+        manager.register_transform::<transform::remove_build::Plugin>(proj, subcommand)?;
+        manager.register_vault_transform(proj, subcommand)?;
```

Comment:
Looks good!

Just one thought: I feel like the `subcommand` should be stored in the transform's `Context`, and maybe not be directly associated with the project all all. So:

```rust
/// The context in which a plugin is being applied.
#[derive(Debug)]
pub struct Context<'a> {
    /// The project to which we're applying this plugin.
    pub project: &'a Project,
    /// The pod to which we're applying this plugin.
    pub pod: &'a Pod,
    /// PRIVATE. Allow future extensibility without breaking the API.
    _nonexclusive: PhantomData<()>,
}
```

...should get a `subcommand: String` field, and the transform should (1) always be registered, and (2) check the context to see if the transform should do anything for that subcommand. Does this make sense?

Thank you for looking into this; it will be nice to fix the random annoying extra builds that nobody wants!

Reply:
That makes a lot more sense actually. I'll look into it!